torch/optim/swa_utils.py

```diff
@@ -37,8 +37,12 @@ def ema_update(ema_param_list, current_param_list, _):
 def get_swa_multi_avg_fn():
     @torch.no_grad()
     def swa_update(averaged_param_list, current_param_list, num_averaged):
-        diffs = torch._foreach_sub(current_param_list, averaged_param_list)
-        torch._foreach_addcdiv_(averaged_param_list, diffs, [num_averaged + 1] * len(averaged_param_list))
+        # foreach lerp only handles float and complex
+        if torch.is_floating_point(averaged_param_list[0]) or torch.is_complex(averaged_param_list[0]):
+            torch._foreach_lerp_(averaged_param_list, current_param_list, 1 / (num_averaged + 1))
+        else:
```

Comment:
btw, should support for this else clause be a feature request for foreach_lerp?

Reply:
Maybe, but this doesn't seem that useful. Most parameters we work on are floats

Reply:
Yea, would be good to add an issue to track if there isn't one already

Reply:
Yea, I also wondered why at all these try to handle integer parameters / buffers given that largely PyTorch doesn't support non-float weights at all (are they supported at all by the regular optimizers?). To support quantized weights?

Reply:
Now that I think about this...lerp for integers would be...an interesting thing to define but not really what we defined lerp to be. And to @vadimkantorov's point, our regular optims today do not support integers since our grads are not integers, so yea, this wouldn't make too big of any difference.

Reply:
Well, I think that torch.lerp is a useful thing to support for uint8/int16: https://github.com/pytorch/pytorch/issues/54389

But I created that feature request outside of the optimizer context - more for efficient image / audio blending

Reply:
And in this context, I would think that just an `assert all(torch.is_floating_point(param) or torch.is_complex(param) for param in averaged_param_list)` and no integer code path should be sufficient...

Reply:
I think keeping support for this `else` case is fine, in case the user throws in all of the params+buffer into SWA or EMA, and that includes the int buffer of BN or some custom buffer, we don't want it to crash.

Reply:
Okay, I extended the feature request https://github.com/pytorch/pytorch/issues/54389

Please feel free to upvote :)